Networking/NetworkingTests/Remote/ProductsRemoteTests.swift

```diff
@@ -89,6 +89,34 @@ class ProductsRemoteTests: XCTestCase {
         XCTAssertEqual(product.productID, sampleProductID)
     }
 
+    /// Verifies that loadProduct properly parses the `product-external` sample response.
+    ///
+    func testLoadSingleExternalProductProperlyReturnsParsedProduct() throws {
+        // Given
+        let remote = ProductsRemote(network: network)
+        let expectation = self.expectation(description: "Load single external product")
+
+        network.simulateResponse(requestUrlSuffix: "products/\(sampleProductID)", filename: "product-external")
+
+        // When
+        var resultMaybe: Result<Product, Error>?
+        remote.loadProduct(for: sampleSiteID, productID: sampleProductID) { aResult in
+            resultMaybe = aResult
+            expectation.fulfill()
+        }
+
+        wait(for: [expectation], timeout: Constants.expectationTimeout)
+
+        // Then
+        let result = try XCTUnwrap(resultMaybe)
+        XCTAssertTrue(result.isSuccess)
+
+        let product = try result.get()
+        XCTAssertEqual(product.productID, sampleProductID)
+        XCTAssertEqual(product.buttonText, "Hit the slopes")
+        XCTAssertEqual(product.externalURL, "https://snowboarding.com/product/rentals/")
```

Comment:
I suggest here to use the new callback `waitForExpectation:`

```suggestion
        let remote = ProductsRemote(network: network)

        network.simulateResponse(requestUrlSuffix: "products/\(sampleProductID)", filename: "product-external")

        // When
        var resultMaybe: Result<Product, Error>?
        waitForExpectation { (exp) in
            remote.loadProduct(for: sampleSiteID, productID: sampleProductID) { aResult in
                resultMaybe = aResult
                exp.fulfill()
            }
        }

        // Then
        let result = try XCTUnwrap(resultMaybe)
        XCTAssertTrue(result.isSuccess)

        let product = try result.get()
        XCTAssertEqual(product.productID, sampleProductID)
        XCTAssertEqual(product.buttonText, "Hit the slopes")
        XCTAssertEqual(product.externalURL, "https://snowboarding.com/product/rentals/")
```

Reply:
updated in https://github.com/woocommerce/woocommerce-ios/pull/2392/commits/d78a26ed188f0d47b5c37d13f48e7c6278521b61 :)